subsys/net/openthread/rpc/server/ot_rpc_ctrl_server.c

```diff
@@ -544,6 +547,142 @@ static void ot_rpc_cmd_get_factory_assigned_eui64(const struct nrf_rpc_group *gr
 	nrf_rpc_cbor_rsp_no_err(group, &rsp_ctx);
 }
 
+static void ot_rpc_cmd_netdata_get(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+				   void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+	otError error = OT_ERROR_NONE;
+	bool stable;
+	bool decoded_ok;
+	uint8_t length = 0;
+
+	NRF_RPC_CBOR_ALLOC(group, rsp_ctx, sizeof(error) + length + 3);
```

Comment:
Why allocating the response context before validating the request? `length` is not yet decoded in here. Also you would need to discard the context on the request decode error not to leak memory.